### PR TITLE
Misc fixes and improvements

### PR DESCRIPTION
**Pull request by mvieth**
See individual commit messages for more information

### CODE REVIEW

File: segmentation/include/pcl/segmentation/ground_plane_comparator.h [hunk 1]
```diff
@@ -213,14 +213,15 @@ namespace pcl
       {
         // Normal must be similar to neighbor
         // Normal must be similar to expected normal
-        float threshold = distance_threshold_;
-        if (depth_dependent_)
-        {
-          Eigen::Vector3f vec = (*input_)[idx1].getVector3fMap ();
+        // TODO check logic in this class: which member variables are needed?
```
Comment (larshg) [inline segmentation/include/pcl/segmentation/ground_plane_comparator.h:216]: Open an issue rather than a note that will be forgotten 😀 ?

**Review by larshg — APPROVED** [review]

**Comment by larshg** [comment]
Should we require 3.10 as we do in main repo for all tutorials. I think the installed config file also require 3.10 when consuming PCL.

File: segmentation/include/pcl/segmentation/ground_plane_comparator.h (hunk 1 above)
Comment (mvieth) [reply segmentation/include/pcl/segmentation/ground_plane_comparator.h:216]: Fair point, there is already issue https://github.com/PointCloudLibrary/pcl/issues/5099 to remind us to check the logic in the class

**Comment by mvieth** [comment]
> Should we require 3.10 as we do in main repo for all tutorials. I think the installed config file also require 3.10 when consuming PCL

As long as the tutorials themselves don't need anything newer than CMake 3.5, I would leave it at 3.5. I updated it because compatibility with CMake < 2.8.12 will be removed in future CMake versions, and the other tutorials require 3.5, so it's nice to have the same required version for all tutorials.

File: segmentation/include/pcl/segmentation/ground_plane_comparator.h (hunk 1 above)
Comment (larshg) [reply segmentation/include/pcl/segmentation/ground_plane_comparator.h:216]: Yeah, thats fine 😀

**Pull request merged**